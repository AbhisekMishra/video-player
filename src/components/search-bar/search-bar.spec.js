import React from 'react';
import { shallow, configure } from 'enzyme';
import Component from './search-bar';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('SearchBar', () => {
    let props;
    let getShallowComponent;

    beforeEach(() => {
        props = {
            classes: {},
            setLink: jest.fn(),
            link: '',
            renderVideo: jest.fn(),
        };

        getShallowComponent = () => shallow(<Component {...props} />).dive();
    });

    describe('render', () => {
        it('should render with all props', () => {
            expect(getShallowComponent()).toMatchSnapshot();
        });

        it('snapshot with instance', () => {
            expect(getShallowComponent().instance()).toMatchSnapshot();
        });
    });

    describe('TextField', () => {
        it('should call props.setLink on change', () => {
            const event = {
                target: {
                    value: 'abc',
                },
            };
            getShallowComponent().find('TextField').simulate('change', event);
            expect(props.setLink).toBeCalled();
        });
    });

    describe('Button', () => {
        it('should call props.renderVideo on click', () => {
            getShallowComponent().find('.SearchBar-button-3').simulate('click');
            expect(props.renderVideo).toBeCalled();
        });
    });
});
