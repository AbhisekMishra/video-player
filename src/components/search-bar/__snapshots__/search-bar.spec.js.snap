// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar render should render with all props 1`] = `ShallowWrapper {}`;

exports[`SearchBar render snapshot with instance 1`] = `
SearchBar {
  "context": Object {},
  "props": Object {
    "classes": Object {
      "button": "SearchBar-button-3",
      "header": "SearchBar-header-2",
      "textField": "SearchBar-textField-1",
    },
    "link": "",
    "renderVideo": [MockFunction],
    "setLink": [MockFunction],
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <SearchBar
        classes={
          Object {
            "button": "SearchBar-button-3",
            "header": "SearchBar-header-2",
            "textField": "SearchBar-textField-1",
          }
        }
        link=""
        renderVideo={[MockFunction]}
        setLink={[MockFunction]}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div>
        <WithStyles(Grid)
          alignItems="flex-end"
          container={true}
          justify="center"
          spacing={16}
        >
          <WithStyles(Grid)
            item={true}
            md={3}
            sm={11}
            xs={11}
          >
            <h3
              className="SearchBar-header-2"
            >
              Enter video URL
            </h3>
            <TextField
              className="SearchBar-textField-1"
              id="standard-search"
              label="Link"
              margin="normal"
              onChange={[Function]}
              required={false}
              select={false}
              type="search"
              value=""
              variant="standard"
            />
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            md={1}
            sm={2}
            xs={2}
          >
            <WithStyles(Button)
              className="SearchBar-button-3"
              color="primary"
              onClick={[Function]}
              variant="contained"
            >
              Go
            </WithStyles(Button)>
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
